template.yaml

```diff
@@ -612,25 +627,64 @@ Resources:
                 - !GetAtt ClientSecretRetrieval.ClientSecret
                 - ""
 
+  LambdaCodeUpdateHandlerRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - edgelambda.amazonaws.com
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies: 
+        - PolicyName: MyCustomPolicy2
+          PolicyDocument: 
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - lambda:GetFunction
+                  - lambda:UpdateFunctionCode
+                Resource:
+                  - !GetAtt ParseAuthHandler.Arn
+                  - !GetAtt CheckAuthHandler.Arn
+                  - !GetAtt HttpHeadersHandler.Arn
+                  - !GetAtt RefreshAuthHandler.Arn
+                  - !GetAtt SignOutHandler.Arn
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+
+  MyCustomPolicy: 
+    Type: AWS::IAM::Policy
+    Properties: 
+      PolicyDocument: 
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - lambda:GetFunction
+              - lambda:UpdateFunctionCode
+            Resource:
+              - !GetAtt ParseAuthHandler.Arn
+      PolicyName: MyCustomPolicy1
+      Roles: 
+        - !Ref LambdaCodeUpdateHandlerRole
+
+
+
+
   LambdaCodeUpdateHandler:
     Type: AWS::Serverless::Function
     Properties:
       CodeUri: src/cfn-custom-resources/lambda-code-update/
       Handler: index.handler
       Runtime: nodejs12.x
-      Policies:
-        - Version: "2012-10-17"
-          Statement:
-            - Effect: Allow
-              Action:
-                - lambda:GetFunction
-                - lambda:UpdateFunctionCode
-              Resource:
-                - !GetAtt ParseAuthHandler.Arn
-                - !GetAtt CheckAuthHandler.Arn
-                - !GetAtt HttpHeadersHandler.Arn
-                - !GetAtt RefreshAuthHandler.Arn
-                - !GetAtt SignOutHandler.Arn
```

Comment:
What's the benefit of your more elaborate policy/role implementation versus this basic one?

Reply:
As opposed to inline IAM policies, having an external policy allows other cf templates to reuse the policy and in this care it gives more flexibility for cases when the developer wants to extend the number of lambda functions written (see knowledge-vault case where i'm adding one more function). If this policy gets too relaxed, it would have too much power since it can update functions's code.